kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSClassDeclarationImpl.kt

```diff
@@ -25,7 +26,14 @@ import org.jetbrains.kotlin.analysis.api.symbols.*
 import org.jetbrains.kotlin.psi.KtClassOrObject
 import org.jetbrains.kotlin.psi.KtObjectDeclaration
 
-class KSClassDeclarationImpl(private val ktNamedClassOrObjectSymbol: KtNamedClassOrObjectSymbol) : KSClassDeclaration {
+class KSClassDeclarationImpl private constructor(
+    private val ktNamedClassOrObjectSymbol: KtNamedClassOrObjectSymbol
+) : KSClassDeclaration {
+    companion object : KSObjectCache<KtNamedClassOrObjectSymbol, KSClassDeclarationImpl>() {
+        fun getCached(ktNamedClassOrObjectSymbol: KtNamedClassOrObjectSymbol) = cache
```

Comment:
Since repeated, I thought I was wrong, but according to https://developer.android.com/kotlin/style-guide#where_to_break, "prefer to break at a higher syntactic level." Here, function declaration is a higher than single expression body I think.